unitOneAssessmentPractice.js

```diff
@@ -3,75 +3,146 @@ let assert = require('assert')
 // Question One:
 
 // Write a function called double that doubles a number
-
+const double = (num) => {
+  return num + num
+}
 // Uncomment out the next line to test your solution
-// runQ1Tests()
+runQ1Tests()
 
 // Question Two:
 
 // Write a function called containsSeven that returns whether or not a number has an sevens in it.
-
+  const containsSeven = (num) => {
+    num = num.toString()
+    return num.includes(7)
+  }
 // Uncomment out the next line to test your solution
-// runQ2Tests()
+runQ2Tests()
 
 // Question Three:
 
 // Write a function called capitalizeTheAs the capitalizes all of the lowercase "a"s in a string leaving all other characters the same
-
+  const capitalizeTheAs = (str) => {
+    newStr =""
+    for (char of str){
+      if (char === "a") {
+        newStr += char.toUpperCase()
+      } else {
+        newStr += char
+      }
+    }
+    return newStr
+  }
 // Uncomment out the next line to test your solution
-// runQ3Tests()
+runQ3Tests()
 
 // Question Four:
 // Write a function called largest that returns the largest value in an array
-
+  const largest = (arr) => {
+    let num = -Infinity
+    arr.forEach((el) => {
+      if (el > num) {
+        num = el
+      }
+    })
+    return num
+  }
 // Uncomment out the next line to test your solution
-// runQ4Tests()
+runQ4Tests()
 
 // Question Five:
 // Write a function called average that returns the average of an array
-
+  const average = (arr) => {
+    let sum = 0
+    arr.forEach(el => sum += el)
+    return sum / arr.length
+  }
 // Uncomment out the next line to test your solution
-// runQ5Tests()
+runQ5Tests()
 
 // Question Six:
 // Write a function called mode that returns the most frequently occurring number in an array
 // HINT: Use an object where the keys are the numbers, and the values are how many times they appear in the array.
-
+  const mode = (arr) => {
+    let occurences = {}
+    let mostFreq;
+    let num = -Infinity
+    arr.forEach((el) => {
+      if (occurences[el]) {
+        occurences[el]++
+      } else {
+        occurences[el] = 1
+      }
+    })
+    for (let key in occurences){
+      if (occurences[key] > num){
+
+        num = occurences[key]
+        mostFreq = Number(key)
+      }
+    }
+    return mostFreq
+  }
 // Uncomment out the next line to test your solution
-// runQ6Tests()
+runQ6Tests()
 
 // Question Seven: (BONUS)
 // Write a function called median that returns the most median number in an array
 // HINT: You'll need to sort the array first
 // This one's also a bit tricky, feel free to skip it and come back to it.
+  const median = (arr) => {
+    arr.sort((a,b) => a-b)
+    return arr.length % 2 ? arr[Math.floor(arr.length / 2)] : (arr[arr.length / 2] + arr[(arr.length / 2) - 1]) / 2
```

Comment:
I would suggest having a variable for the mid index, this would make it easier to read your code.